.github/workflows/jira.yaml

```diff
@@ -0,0 +1,44 @@
+# this workflow requires to provide JIRA webhook URL via JIRA_URL GitHub Secret
+# read more: https://support.atlassian.com/cloud-automation/docs/jira-automation-triggers/#Automationtriggers-Incomingwebhook
+# original code source: https://github.com/beliaev-maksim/github-to-jira-automation
+
+name: Issues to JIRA
+
+on:
+  workflow_call:
+
+jobs:
+  update:
+    name: Update Issue
+    runs-on: ubuntu-latest
+    steps:
+      - name: Create JIRA ticket
+        env:
+          # put into env vars to properly escape special bash chars
+          ISSUE_TITLE: ${{ github.event.issue.title }}
+          ISSUE_DESCRIPTION: ${{ github.event.issue.body }}
+        run: |
+          # Determine Bug/Enhancement
+          if ${{ contains(github.event.*.labels.*.name, 'Bug') }}; then
+            type=bug
+          else
+            type=story
+          fi
+
+          # send JIRA request
+
+          # Canonical discourages discussions in JIRA on public project issues.
+          # It is NOT recommended to put description in JIRA in order to
+          # push conversation to GitHub. If strongly required,
+          # replace '--arg body ""' with '--arg body "$ISSUE_DESCRIPTION"'
+
+          data=$( jq -n \
+                  --arg title "$ISSUE_TITLE" \
+                  --arg url '${{ github.event.issue.html_url }}' \
+                  --arg submitter '${{ github.event.issue.user.login }}' \
```

Comment:
Will this be ok if the submitter is someone external to Canonical and therefore won't have an account on Jira ?

Reply:
I didn't even think about that, thanks for raising this! I did a manual curl with a bogus user - the Reporter of the issue in Jira is "Automation for Jira", the `submitter` is used only in the description field when creating the story, together with a link to the original issue. It is for informational purposes, so it works with all submitters.

Reply:
Yeap so it uses the user whose API token is attached to